presto-native-execution/presto_cpp/main/operators/LocalShuffle.cpp

```diff
@@ -305,6 +352,43 @@ void LocalShuffleWriter::noMoreData(bool success) {
   }
 }
 
+SortedFileInputStream::SortedFileInputStream(
+    const std::string& filePath,
+    uint16_t streamIdx,
+    velox::memory::MemoryPool* pool,
+    size_t bufferSize)
+    : velox::common::FileInputStream(
+          velox::filesystems::getFileSystem(filePath, nullptr)
+              ->openFileForRead(filePath),
+          bufferSize,
+          pool),
+      filePath_(filePath),
+      streamIdx_(streamIdx) {
+  // Initialize by reading first row.
+  next();
+}
+
+bool SortedFileInputStream::next() {
+  if (atEnd()) {
+    currentKey_.clear();
+    currentData_.clear();
+    return false;
+  }
+  const auto keySize = folly::Endian::big(read<TRowSize>());
+  const auto dataSize = folly::Endian::big(read<TRowSize>());
+
+  currentKey_.resize(keySize);
+  if (keySize > 0) {
+    readBytes(reinterpret_cast<uint8_t*>(currentKey_.data()), keySize);
```

Comment:
maybe we can use nextView to avoid the copy. But be careful of taking care of the edge cases at range boundary.